apps/backend/src/application/application.service.ts

```diff
@@ -0,0 +1,139 @@
+            },
+          },
+          applicationPeriod: {
+            connect: {
+              id: createApplicationDto.applicationPeriodId,
+            },
+          },
+        },
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2002') {
+          throw new BadRequestException('Ez a jelentkezés már létezik');
+        } else if (e.code === 'P2025') {
+          throw new NotFoundException('Nem található időszak');
+        }
+      }
+      if (e instanceof BadRequestException) {
+        throw e;
+      }
+      throw new BadRequestException('Nem sikerült létrehozni');
+    }
+  }
+
+  @UseGuards(AuthGuard('jwt'), RolesGuard)
+  @ApiBearerAuth()
+  @Roles(Role.BODY_ADMIN, Role.BODY_MEMBER)
+  async findAll() {
+    return await this.prisma.application.findMany();
+  }
+
+  async findOne(id: number): Promise<Application> {
+    const application = this.prisma.application.findFirst({
+      where: {
+        id,
+      },
+    });
+    if (!application) {
+      throw new NotFoundException('A keresett jelentkezés nem található');
+    }
+    return application;
+  }
+
+  async getCurrentUserApplication(user: User): Promise<Application | null> {
+    const currentPeriod = await this.applicationPeriodService.getCurrentPeriod();
+    try {
+      return await this.prisma.application.findFirstOrThrow({
+        where: {
+          AND: [
+            {
+              applicationPeriodId: currentPeriod.id,
+            },
+            {
+              userId: user.authSchId,
+            },
+          ],
+        },
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2025') {
+          throw new NotFoundException('Nem található jelentkezés');
+        }
+      }
+    }
+  }
+
+  async update(id: number, updateApplicationDto: UpdateApplicationDto): Promise<Application> {
+    try {
+      return await this.prisma.application.update({
+        where: {
+          id,
+        },
+        data: {
+          status: updateApplicationDto.applicationStatus,
+        },
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2025') {
+          throw new NotFoundException('A keresett jelentkezés nem található');
+        }
+      }
+    }
+  }
+
+  async remove(id: number, user: User): Promise<Application> {
+    const application = await this.prisma.application.findFirst({
+      where: {
+        id,
+      },
+    });
+    if (!application) {
+      throw new NotFoundException('A keresett jelentkezés nem található');
+    }
+    if (application.userId === user.authSchId || user.role === Role.BODY_ADMIN || user.role === Role.BODY_MEMBER) {
+      return await this.prisma.application.delete({
+        where: {
+          id,
+        },
+      });
+    }
+    throw new ForbiddenException('Nem törölheted mások jelentkezését');
+  }
```

Comment:
should the user be able to delete their application after the application period has ended? that could cause some confusion for the body members